docs/r2/image_summaries.ipynb

```diff
@@ -0,0 +1,639 @@
+        "  cm = np.around(cm.astype('float') / cm.sum(axis=1)[:, np.newaxis], decimals=2)\n",
+        "\n",
+        "  # Use white text if squares are dark; otherwise black.\n",
+        "  threshold = cm.max() / 2.\n",
+        "  for i, j in itertools.product(range(cm.shape[0]), range(cm.shape[1])):\n",
+        "    color = \"white\" if cm[i, j] > threshold else \"black\"\n",
+        "    plt.text(j, i, cm[i, j], horizontalalignment=\"center\", color=color)\n",
+        "\n",
+        "  plt.tight_layout()\n",
+        "  plt.ylabel('True label')\n",
+        "  plt.xlabel('Predicted label')\n",
+        "  return figure"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "6lOAl_v26QGq",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "You're now ready to train the classifier and regularly log the confusion matrix along the way.\n",
+        "\n",
+        "Here's what you'll do:\n",
+        "\n",
+        "1. Create the Keras TensorBoard callback to log basic metrics\n",
+        "2. Create a Keras LambdaCallback to log the confusion matrix at the end of every epoch\n",
+        "3. Train the model using Model.fit(), making sure to pass both callbacks\n",
+        "\n",
+        "As training progresses, scroll down to see TensorBoard start up."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "bXQ7-9CF0TPA",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "# Clear out prior logging data.\n",
+        "!rm -rf logs/image\n",
+        "\n",
+        "logdir = \"logs/image/\" + datetime.now().strftime(\"%Y%m%d-%H%M%S\")\n",
+        "# Define the basic TensorBoard callback.\n",
+        "tensorboard_callback = keras.callbacks.TensorBoard(log_dir=logdir)\n",
+        "file_writer_cm = tf.summary.create_file_writer(logdir + '/cm')\n",
+        "\n",
+        "def log_confusion_matrix(epoch, logs):\n",
+        "  # Use the model to predict the values from the validation dataset.\n",
+        "  test_pred_raw = model.predict(test_images)\n",
+        "  test_pred = np.argmax(test_pred_raw, axis=1)\n",
+        "\n",
+        "  # Calculate the confusion matrix.\n",
+        "  cm = sklearn.metrics.confusion_matrix(test_labels, test_pred)\n",
+        "  # Log the confusion matrix as an image summary.\n",
+        "  figure = plot_confusion_matrix(cm, class_names=class_names)\n",
+        "  cm_image = plot_to_image(figure)\n",
+        "\n",
+        "  # Log the confusion matrix as an image summary.\n",
+        "  with file_writer_cm.as_default():\n",
+        "    tf.summary.image(\"Confusion Matrix\", cm_image, step=epoch)\n",
+        "\n",
+        "# Define the per-epoch callback.\n",
+        "cm_callback = keras.callbacks.LambdaCallback(on_epoch_end=log_confusion_matrix)\n",
+        "\n",
+        "%tensorboard --logdir logs/image\n",
```

Comment:
It might be worth splitting these into different cells, especially for better rendering in the docs website.

Reply:
Done.